sdk/nodejs/cmd/pulumi-language-nodejs/main.go

```diff
@@ -1969,10 +1971,99 @@ func (o *oomSniffer) Scan(r io.Reader) {
 func (host *nodeLanguageHost) Link(
 	ctx context.Context, req *pulumirpc.LinkRequest,
 ) (*pulumirpc.LinkResponse, error) {
-	// The NodeJS language host does not support linking, so we return an empty response.
-	return &pulumirpc.LinkResponse{}, nil
+	logging.V(5).Infof("Linking %+v in %s", req.Packages, req.Info.RootDirectory)
+	opts, err := parseOptions(req.Info.Options.AsMap())
+	if err != nil {
+		return nil, err
+	}
+
+	loader, err := schema.NewLoaderClient(req.LoaderTarget)
+	if err != nil {
+		return nil, err
+	}
+	defer loader.Close()
+	cachedLoader := schema.NewCachedLoader(loader)
+
+	pm, err := npm.ResolvePackageManager(opts.packagemanager, req.Info.ProgramDirectory)
+	if err != nil {
+		return nil, err
+	}
+
+	instructions := "You can import the SDK in your JavaScript code with:\n\n"
+	if opts.typescript {
```

Comment:
I think that typescript might be false but the project might still be a typescript project. typescript just sets us up to run with ts-node but someone might be using tsc.

Reply:
Right. This check mirrors the existing one that also looked at the options.

I think we could look for the presence of a tsconfig.json file, so check for `opts.typescript || hasTSConfig`. There's likely some cases where that is not entirely right, but there's a humungous amount of different options that will determine this, and some likely undetectable by us.